buildSrc/src/main/java/Dependencies.kt

```diff
@@ -1,5 +1,5 @@
 object BuildPluginsVersion {
-    val AGP = System.getenv("VERSION_AGP") ?: "4.2.2"
+    val AGP = System.getenv("VERSION_AGP") ?: "7.0.1"
```

Comment:
this change would be ideally done with #109 since there are new APIs, also requires bumping a major version of the plugin

Reply:
Yeah, but not sure I want to switch to those other changes rn. I just did it for the purpose of trying out the new api, which turned out to be working :) Maybe I'll finish the instrumentation first, and then if it's not tackled by then, I will implement #109 properly? This branch is anyway targeting an integration branch and no `main`

Reply:
yep, that sounds great, but we have to work on #109 before releasing it, otherwise, we'd be mixing deprecated APIs and non-stable APIs in the same version, not the best combination.
obviously this could be done in another PR at a later point

Reply:
So are we doing the bump to AGP 7 in this PR?

Reply:
yes and no, this PR merges to the feat/bytecode-instrumentation branch and @romtsn will work on #109 in parallel, since the bytecode man. depends on #109

Reply:
Exactly, will open a new PR to `main` with just the bump and later follow up with another PRs to switch to new APIs if not done by that time yet